File: src/main/java/com/movierental/Rentals.java
package com.movierental;

import java.util.ArrayList;


public class Rentals extends ArrayList<Rental> {

    double totalAmount(Rentals rentals) {
        return rentals.stream().map(rental -> rental.amount()).reduce(0.0D,Double::sum);
    }

    int totalFrequentRenterPoints(Rentals rentals) {
        return rentals.stream().map(rental -> rental.FrequentRenterPoints()).reduce(0,Integer::sum);
    }
}

File: src/main/kotlin/com/cleancode/kotlin/Rentals.kt
package com.cleancode.kotlin
class Rentals : ArrayList<Rental>() {
    fun totalAmount(rentals: Rentals): Double {
        return rentals.map{rental->rental.amount()}.sum()
    }

    fun totalFrequentRenterPoints(rentals: Rentals): Int {
        return rentals.map{rental->rental.FrequentRenterPoints()}.sum()
    }
}